one_fm/one_fm/hr_utils.py

```diff
@@ -9,179 +9,182 @@
+                        "allocated_amount": doc.get("outstanding_amount"),
+                    },
+                )
+                pe.append(
+                    "references",
+                    {
+                        "reference_doctype": dt,
+                        "reference_name": dn,
+                        "bill_no": doc.get("bill_no"),
+                        "due_date": doc.get("due_date"),
+                        "total_amount": doc.get("dunning_amount"),
+                        "outstanding_amount": doc.get("dunning_amount"),
+                        "allocated_amount": doc.get("dunning_amount"),
+                    },
+                )
+            else:
+                pe.append(
+                    "references",
+                    {
+                        "reference_doctype": dt,
+                        "reference_name": dn,
+                        "bill_no": doc.get("bill_no"),
+                        "due_date": doc.get("due_date"),
+                        "total_amount": grand_total,
+                        "outstanding_amount": outstanding_amount,
+                        "allocated_amount": outstanding_amount,
+                    },
+                )
+
+    pe.setup_party_account_field()
+    pe.set_missing_values()
+
+    update_accounting_dimensions(pe, doc)
+
+    if party_account and bank:
+        pe.set_exchange_rate(ref_doc=reference_doc)
+        pe.set_amounts()
+        if discount_amount:
+            pe.set_gain_or_loss(
+                account_details={
+                    "account": frappe.get_cached_value("Company", pe.company, "default_discount_account"),
+                    "cost_center": pe.cost_center
+                                   or frappe.get_cached_value("Company", pe.company, "cost_center"),
+                    "amount": discount_amount * (-1 if payment_type == "Pay" else 1),
+                }
+            )
+            pe.set_difference_amount()
+
+    return pe
 
 def validate_leave_proof_document_requirement(doc, method):
-	'''
-		Function to validate Is Proof Document Required Flag in Leave Application
-		Triger form Validate hook of Leave Application
-	'''
-
-	if doc.leave_type and doc.status in ['Open', 'Approved'] and not doc.is_new():
-		doc.is_proof_document_required = frappe.db.get_value('Leave Type', doc.leave_type, 'is_proof_document_required')
-		if doc.is_proof_document_required and not doc.proof_documents:
-			frappe.throw(_("Proof Document Required for {} Leave Type".format(doc.leave_type)))
-
-		for each in doc.proof_documents:
-			if not each.attachments:
-				frappe.throw(_("Proof Document Required for {} Leave Type in row {}".format(doc.leave_type,each.idx)))
+    '''
+        Function to validate Is Proof Document Required Flag in Leave Application
+        Triger form Validate hook of Leave Application
+    '''
+    check_status = frappe.db.get_value("Employee", doc.employee, "non_kuwaiti_residents")
```

Comment:
@NoBoneZ I think if you are creating a new field to represent  non kuwaiti residents, You should include a patch that updates the employee record on staging and production when this is approved.

Reply:
Ok, i will do that